packages/@aws-cdk/pipelines/README.md

```diff
@@ -62,6 +107,524 @@ class MyApplication extends Stage {
+of the pipeline development workflow. Each change to the pipeline must be pushed
+to git, otherwise, after the pipeline was updated using `cdk deploy`, it will
+automatically revert to the state found in git.
+
+To make the development more convenient, the self-mutation feature can be turned
+off temporarily, by passing `selfMutation: false` property, example:
+
+```ts
+// Modern API
+const pipeline = new CodePipeline(this, 'Pipeline', {
+  selfMutation: false,
+  ...
+});
+
+// Original API
+const pipeline = new CdkPipeline(this, 'Pipeline', {
+  selfMutating: false,
+  ...
+});
+```
+
+## Definining the pipeline - Modern API
+
+This section of the documentation describes the AWS CodePipeline engine, which
+comes with this library. If you want to use a different deployment engine, read
+the section *Using a different deployment engine* below.
+
+### Synth and sources
+
+To define a pipeline, instantiate a `CodePipeline` construct. It takes
+one argument, a `synth` step, which is expected to produce the CDK Cloud
+Assembly as its single output (the contents of the `cdk.out` directory after
+running `cdk synth`). "Steps" are arbitrary actions in the pipeline, typically
+used to run scripts or commands.
+
+For the synth, use a `ScriptStep` and specify the commands necessary to build
+your project and run `cdk synth`; the specific commands required will depend on
+the programming language you are using. For a typical NPM-based project, the synth
+will look like this:
+
+```ts
+const source = /* the repository source */;
+
+const pipeline = new CodePipeline(this, 'Pipeline', {
+  synth: new ScriptStep('Synth', {
+    input: source,
+    commands: [
+      'npm ci',
+      'npm run build',
+      'npx cdk synth',
+    ],
+  }),
+});
+```
+
+The pipeline assumes that your `ScriptStep` will produce a `cdk.out`
+directory in the root, containing the CDK cloud assembly. If your
+CDK project lives in a subdirectory, be sure to adjust the
+`primaryOutputDirectory` to match:
+
+```ts
+const pipeline = new CodePipeline(this, 'Pipeline', {
+  synth: new ScriptStep('Synth', {
+    input: source,
+    commands: [
+      'cd mysubdir',
+      'npm ci',
+      'npm run build',
+      'npx cdk synth',
+    ],
+    primaryOutputDirectory: 'mysubdir/cdk.out',
+  }),
+});
+```
+
+The underlying CodePipeline `Pipeline` construct will be produced
+when `app.synth()` is called. You can also force it to be produced
+earlier by calling `pipeline.buildPipeline()`. After you've called
+that method, you can inspect the constructs that were produced by
+accessing the properties of the `pipeline` object.
+
+#### CodePipeline Sources
+
+In CodePipeline, *Sources* define where the source of your application lives.
+When a change to the source is detected, the pipeline will start executing.
+Sources objects can be created by factory methods on the `CodePipelineSource` class:
+
+* `CodePipelineSource.gitHub('org/repo', { branch, authentication })`: use a
```

Comment:
I still think this should be `github` (just more aesthetic):

```suggestion
* `CodePipelineSource.github('org/repo', { branch, authentication })`: use a
```

(Feel free to dismiss this comment)

Reply:
I just changed it back. I don't disagree with you but in all other places in the library we capitalize the H, and all other CodePipeline and CodeBuild also capitalize the H. 

Felt it was more important to be consistent.

Reply:
Agreed.